packages/create-catalyst/src/commands/integration.ts

```diff
@@ -0,0 +1,118 @@
+}
+
+export const integration = new Command('integration')
+  .requiredOption('--integration-name <name>', 'Formatted name of the integration')
+  .requiredOption('--source-branch <branch>', 'The branch containing your integration source code')
+  .action(async (options) => {
+    const manifest: Manifest = {
+      name: options.integrationName,
+      dependencies: [],
+      devDependencies: [],
+      environmentVariables: [],
+    };
+
+    const { stdout: gitBranchStdOut } = await exec('git branch --format="%(refname:short)"');
+    const localBranches = gitBranchStdOut.split('\n').filter(Boolean);
+
+    if (!localBranches.includes(options.sourceBranch)) {
+      console.error(`Branch "${options.sourceBranch}" does not exist in your local repository.`);
+      process.exit(1);
+    }
+
+    const { stdout: catalystTags } = await exec('git tag --list');
+    const coreTagsSorted = catalystTags
+      .split('\n')
+      .filter((tag) => tag.startsWith('@bigcommerce/catalyst-core@'))
+      .map((tag) => tag.replace('@bigcommerce/catalyst-core@', ''))
+      .map((tag) => semver.coerce(tag))
+      .filter((tag) => tag !== null)
+      .sort(semver.compare)
+      .map((tag) => tag.version);
+
+    const latestCoreTag = `@bigcommerce/catalyst-core@${coreTagsSorted[coreTagsSorted.length - 1]}`;
+
+    const { stdout: packagesDiffStdOut } = await exec(
+      `git diff ${latestCoreTag}...${options.sourceBranch} -- core/package.json`,
+    );
+
+    if (packagesDiffStdOut.length > 0) {
+      const packages: string[] = [];
+      const lines = packagesDiffStdOut.split('\n');
+      const packagePattern = /^\+ {4}"([^"]+)":/;
+
+      lines.forEach((line) => {
+        const match = line.match(packagePattern);
+
+        if (match) {
+          packages.push(match[1]);
+        }
+      });
+
+      if (packages.length > 0) {
+        const { stdout: integrationPackageJsonRaw } = await exec(
+          `git show ${options.sourceBranch}:core/package.json`,
+        );
+
+        const integrationPackageJson = z
+          .object({
+            dependencies: z.object({}).passthrough(),
+            devDependencies: z.object({}).passthrough(),
+          })
+          .parse(JSON.parse(integrationPackageJsonRaw));
+
+        manifest.dependencies = packages.filter((pkg) => integrationPackageJson.dependencies[pkg]);
+        manifest.devDependencies = packages.filter(
+          (pkg) => integrationPackageJson.devDependencies[pkg],
+        );
+      }
+    }
+
+    const { stdout: envVarDiff } = await exec(
+      `git diff ${latestCoreTag}...${options.sourceBranch} -- core/.env.example`,
+    );
+
+    if (envVarDiff.length > 0) {
+      const envVars: string[] = [];
+      const lines = envVarDiff.split('\n');
+      const envPattern = /^\+([A-Z_]+)=/;
+
+      lines.forEach((line) => {
+        const match = line.match(envPattern);
+
+        if (match) {
+          envVars.push(match[1]);
+        }
+      });
```

Comment:
Same thing here. You could `git show` the file and drop it into `dotenv.parse` I believe.

Reply:
This parses out just the new additions - what happens if an integration developer for some reason deletes some other env vars from their example that diverge from main?

Reply:
Unfortunately importing `dotenv.parse` throws:
```
Error: Dynamic require of "fs" is not supported
```

This can be fixed by adding `external: ['dotenv'],` to the `tsup` config, but not sure if that's something we want to do, or if we just keep the regex approach